packages/editor-ui/src/stores/assistant.store.ts

```diff
@@ -594,12 +600,21 @@ export const useAssistantStore = defineStore(STORES.ASSISTANT, () => {
 				},
 				(msg) => onEachStreamingMessage(msg, id),
 				() => onDoneStreaming(id),
-				(e) => handleServiceError(e, id),
+				(e) =>
+					handleServiceError(
+						e,
+						id,
+						async () => await withoutFailedMessages(async () => await sendMessage(chatMessage)),
```

Comment:
Not really happy with the async/awaits here but they're technically correct and I need them for the test cases I think